bee-node/src/plugins/dashboard/config.rs

```diff
@@ -74,16 +76,44 @@ impl DashboardConfigBuilder {
     }
 
     pub fn finish(self) -> DashboardConfig {
+        let multi_addr = self
+            .bind_addresses
+            .unwrap_or_else(|| DEFAULT_BIND_ADDRESSES.parse().unwrap());
+        let address = multi_addr
+            .iter()
+            .find_map(|x| match x {
+                Protocol::Dns(std::borrow::Cow::Borrowed("localhost")) => Some(
+                    ("localhost", 0)
+                        .to_socket_addrs()
+                        .expect("error resolving 'localhost'")
+                        .nth(0)
+                        .unwrap()
+                        .ip(),
+                ),
+                Protocol::Ip4(ip) => Some(IpAddr::V4(ip)),
+                Protocol::Ip6(ip) => Some(IpAddr::V6(ip)),
+                _ => None,
+            })
+            .unwrap_or_else(|| panic!("Unsupported address"));
+
+        let port = multi_addr
+            .iter()
+            .find_map(|x| match x {
+                Protocol::Tcp(port) => Some(port),
+                _ => None,
+            })
```

Comment:
I suggest using `if let` instead of `match` here:
`.find_map(|x| if let Protocol::Tcp(port) { Some(port) } else { None })

Reply:
I guess the same would apply to port initialization in `bee-api/bee-rest-api/src/endpoints/config.rs `?

Reply:
yea, I guess. It's not a real issue, of course. If there were a good reason this `match` would maybe be expanded later at some point, I'ld say keep it. But I feel like this is not going to happen in this case.